src/sentry/models/group.py

```diff
@@ -421,26 +421,30 @@ def update_group_status(
         """For each groups, update status to `status` and create an Activity."""
         from sentry.models import Activity
 
-        to_be_updated = self.filter(id__in=[g.id for g in groups]).exclude(
-            status=status, substatus=substatus
-        )
+        groups_list = []
+        with transaction.atomic(router.db_for_write(Group)):
+            selected_groups = (
+                Group.objects.filter(id__in=[g.id for g in groups])
+                .exclude(status=status, substatus=substatus)
+                .select_for_update()
+            )
 
-        for group in to_be_updated:
-            group.status = status
-            group.substatus = substatus
+            for group in selected_groups:
+                group.status = status
+                group.substatus = substatus
+                groups_list.append(group)
```

Comment:
This is probably just my lack of knowledge about the django ORM, but why do the groups need to be moved from `selected_groups` into a separate list? Is that so the queryset gets materialized? Would it be equivalent to just wrap the `Group.objects...` expression in a list? (Not saying that's necessarily better, just trying to understand what's happening.)

Reply:
Got it, thanks. Perhaps name it `modified_groups`, then, though?